modules/gapi/test/common/gapi_tests_common.hpp

```diff
@@ -855,6 +992,79 @@ class AbsExactVector : public WrappableVector<AbsExactVector<Elem>, Elem>
         return os << "AbsExactVector()";
     }
 };
+
+template<typename Elem, int cn>
+class AvgDiffToleranceVec : public WrappableVec<AvgDiffToleranceVec<Elem, cn>, Elem, cn>
+{
+public:
+    AvgDiffToleranceVec(double tol) : _tol(tol) {}
+    bool operator() (const cv::Vec<Elem, cn> &in1, const cv::Vec<Elem, cn> &in2) const
+    {
+        double abs_err = 0;
+        for (int i = 0; i < cn; i++)
+        {
+            abs_err += static_cast<double>(std::abs(in1[i] - in2[i]))
```

Comment:
There is a little problem here. If you're comparing integer types you should use `std::abs` but for floating point types there is `std::fabs`

Reply:
Rewritten using `cv::norm`